umpleonline/ump/GeometricSystem.ump

```diff
@@ -115,143 +115,154 @@ class NonPolyhedra{
 class Circle{
 	isA CurvedShape;
 	double radius;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(CurvedShape object){/*implementation */}
-	boolean isEqual(CurvedShape object){/*implementation */}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Ellipse {
 	isA CurvedShape;
 	double majorAxis;
 	double minorAxis;
 	Point fFocus;
 	Point gFocus;
-	Point[2] getFoci(){/*implementation */}
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	Line getTangent(){/*implementation */}
-	boolean isLessThan(CurvedShape object){}
-	boolean isEqual(CurvedShape object){}
+	Point[] getFoci(){return new Point[2];}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	Line getTangent(){return new Line(fFocus, gFocus);}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Quadrilateral{
 	isA Polygon;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(Polygon object){}
-	boolean isEqual(Polygon object){}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(Polygon object){return true;}
+	boolean isEqual(Polygon object){return true;}
+	boolean isEqual(CurvedShape object){return false;}
```

Comment:
Originally, I had removed that method from the examples. In this case, I removed it from the Quadrilateral class. However, when I compiled the Umple file, and then tried to compile Quadrilateral.java, the following error was thrown:  
```
JackieLang:geometric_system$ javac Quadrilateral.java 
Quadrilateral.java:8: error: Quadrilateral is not abstract and does not override abstract method isEqual(CurvedShape) in Polygon
public class Quadrilateral extends Polygon
       ^
1 error```